### ISSUE TITLE
Uncaught ReferenceError: define is not defined

### ISSUE THREAD

**Issue by betterliyu** [anastasiia-zolochevska/react-appinsights#33]
I import react-ai in our main.js and use browserify to build and bundle js file. I met this error when I access my site on browser. It is easy to repro.

main.js
`import ReactAI from 'react-appinsights';`

gulpfile.js
```
gulp.task('srcJs', function () {
    return browserify(config.paths.mainJs, { fullPaths: true, debug: debug, })
        .transform("babelify", { /*plugins: ["lodash"],*/ presets: ["es2015", "react"] }) //lodash plugin does not work here since react-bootstrap-table->react-toastr->lodash reference
        .bundle()
        .on('error', logAndEatError)
        .pipe(source('bundle.js'))
        //.pipe(env_prod(streamify(uglify())))
        .pipe(rename('bundle.min.js'))
        .pipe(gulp.dest(config.paths.dist + '/scripts'));
});
```

package.json snippet
```
    "babel-cli": "^6.10.1",
    "babel-plugin-lodash": "^3.2.2",
    "babel-preset-es2015": "^6.6.0",
    "babel-preset-react": "^6.5.0",
    "babelify": "^7.3.0",
    "browserify": ">=11.2.0",
    "react": "^15.3.2",
    "react-appinsights": "^2.0.1",
    "react-dom": "^15.0.2",
    "react-router": "^2.8.1",
```

![image](https://user-images.githubusercontent.com/19713797/51291037-bc3f3980-1a40-11e9-87b3-c2c33af4ecd5.png)

I think it's caused by the applicationinsights-js you depended on. Please refer this [reply](https://github.com/Microsoft/ApplicationInsights-JS/issues/749#issuecomment-454779197) from applicationinsights-js. 
What are you able to do to fix this issue now on your side or on my side?

**Comment by pviotti** [comment]
Yes, it is a well-known packaging issue of Application Insights (see also https://github.com/Microsoft/ApplicationInsights-JS/issues/476 and https://github.com/Microsoft/ApplicationInsights-JS/issues/301 ).
There are some workarounds (see the comments in the threads above), but the suggested solution would be using AppInsights' new SDK, which is supported by `react-appinsights` v3.x, now in beta.
You can install `react-appinsights` v3.x by issuing `npm install --save react-appinsights@beta`.  Its documentation is in the [typescript-rewrite](https://github.com/Azure/react-appinsights/tree/typescript-rewrite) branch.
Let us know if this solves the issue.

**Comment by betterliyu** [comment]
@pviotti Yes, the beta version is ok to me. What is the ETA of v3 release?

**Comment by pviotti** [comment]
v3 will be released after AppInsights's new SDK v1.0 is released: hopefully within one month or so.
